lib/Backend/BackwardPass.cpp

```diff
@@ -4369,59 +4369,6 @@ BackwardPass::TraceBlockUses(BasicBlock * block, bool isStart)
 
 #endif
 
-bool
-BackwardPass::UpdateImplicitCallBailOutKind(IR::Instr *const instr, bool needsBailOutOnImplicitCall)
-{
-    Assert(instr);
-    Assert(instr->HasBailOutInfo());
-
-    IR::BailOutKind implicitCallBailOutKind = needsBailOutOnImplicitCall ? IR::BailOutOnImplicitCalls : IR::BailOutInvalid;
-
-    IR::BailOutKind instrBailOutKind = instr->GetBailOutKind();
-    if (instrBailOutKind & IR::BailOutMarkTempObject)
-    {
-        // Remove the mark temp object bit, as we don't need it after the dead store pass
-        instrBailOutKind &= ~IR::BailOutMarkTempObject;
-        instr->SetBailOutKind(instrBailOutKind);
-
-        if (!instr->GetBailOutInfo()->canDeadStore)
```

Comment:
Merge was actually more wrong than this. We need to change the other version of `UpdateImplicitCallBailOutKind` as well, which is why the unit test is failing.

Reply:
Basically we need to change it to be same is it is in `unreleased/future`